uaclient/cli.py

```diff
@@ -248,21 +251,51 @@ def status_parser(parser):
     return parser
 
 
-@assert_root
-@require_valid_entitlement_name("disable")
-@assert_attached(ua_status.MESSAGE_ENABLE_FAILURE_UNATTACHED_TMPL)
-def action_disable(args, cfg):
+def _perform_disable(entitlement_name, cfg, *, assume_yes):
     """Perform the disable action on a named entitlement.
 
-    @return: 0 on success, 1 otherwise
+    :param entitlement_name: the name of the entitlement to enable
+    :param cfg: the UAConfig to pass to the entitlement
+    :param assume_yes:
+        Assume a yes response for any prompts during service enable
+
+    @return: True on success, False otherwise
     """
-    ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME[args.name]
-    entitlement = ent_cls(cfg, assume_yes=args.assume_yes)
-    ret = 0 if entitlement.disable() else 1
+    ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME[entitlement_name]
+    entitlement = ent_cls(cfg, assume_yes=assume_yes)
+    ret = entitlement.disable()
     cfg.status()  # Update the status cache
     return ret
 
 
+@assert_root
+@require_valid_entitlement_names("disable")
+@assert_attached(ua_status.MESSAGE_ENABLE_FAILURE_UNATTACHED_TMPL)
+def action_disable(args, cfg, **kwargs):
+    """Perform the disable action on a list of entitlements.
+
+    @return: 0 on success, 1 otherwise
+    """
+    entitlements_found = kwargs.get("entitlements_found", [])
+    entitlements_not_found = sorted(set(args.names) - set(entitlements_found))
+    tmpl = ua_status.MESSAGE_INVALID_SERVICE_OP_FAILURE_TMPL
+    ret = True
+
+    for entitlement in entitlements_found:
```

Comment:
I think both action_disable and action_enable can look kindof like the following:
```suggestion

@assert_root
@assert_attached(ua_status.MESSAGE_ENABLE_FAILURE_UNATTACHED_TMPL)
def action_disable(args, cfg, **kwargs):
...
valid_entitlements = get_valid_entitlements(args.name, allow_beta=args.beta)
entitlement_not_found = sorted(set(args.names) - set(entitlements_found))
ret = True
for ent_name in valid_entitlements:
   ret &= _perform_disable(entitlement, cfg, assume_yes=args.assume_yes)

    if entitlements_not_found:
        raise exceptions.UserFacingError(
            tmpl.format(
                operation="enable", name=", ".join(entitlements_not_found)
            )
        )
 return 0 if ret else 1
```